File: phaseGateOne/BankeBank/CreditCardValidator.java
import java.util.Scanner;

public class CreditCardValidator {
  
  
    public static boolean isValidLength(String cardNumber) {
        int length = cardNumber.length();
        return length >= 13 && length <= 16;
    }

  public static String determineCardType(String cardNumber) {
    if (cardNumber.startsWith("4")) {
      return "Visa Card";
    } else if (cardNumber.startsWith("5")) {
      return "MasterCard";
    } else if (cardNumber.startsWith("37")) {
      return "American Express Card";
    } else if (cardNumber.startsWith("6")) {
      return "Discover Card";
    }
    return "Invalid Card Type";
  }
  
  public static boolean isValidCard(String cardNumber) {
    int sumOfDoubleEvenPlace = sumOfDoubleEvenPlace(cardNumber);
    int sumOfOddPlace = sumOfOddPlace(cardNumber);
    int totalSum = sumOfDoubleEvenPlace + sumOfOddPlace;
    
    return totalSum % 10 == 0;
  }
  
  public static int sumOfDoubleEvenPlace(String cardNumber) {
    int sum = 0;
    for (int i = cardNumber.length() - 2; i >= 0; i -= 2) {
      int digit = Character.getNumericValue(cardNumber.charAt(i));
      int doubledDigit = digit * 2;
      sum += doubledDigit > 9 ? (doubledDigit / 10 + doubledDigit % 10) : doubledDigit;
    }
    return sum;
  }
    
    public static int sumOfOddPlace(String cardNumber) {
      int sum = 0;
      for (int i = cardNumber.length() - 1; i >= 0; i -= 2) {
        sum += Character.getNumericValue(cardNumber.charAt(i));
      }
      return sum;
  }
}

File: phaseGateOne/BankeBank/creditcardvalidator.py
def isValidLength(cardNumber):
    length = cardNumber.length();
    return length >= 13 && length <= 16;
  

def determineCardType(cardNumber):
    if (cardNumber.startsWith("4")):
        return "Visa Card";
    elif (cardNumber.startsWith("5")):
        return "MasterCard";
    elif (cardNumber.startsWith("37")):
        return "American Express Card";
    elif (cardNumber.startsWith("6")):
        return "Discover Card";

    return "Invalid Card Type";
  
  
def sumOfDoubleEvenPlace(cardNumber) {
    total = 0;
    for i in range(len(cardNumber) - 2, 0, -2):
        digit = cardNumber[i];
        doubledDigit = digit * 2;
        total += (doubledDigit // 10 + doubledDigit % 10) if doubledDigit > 9 else doubledDigit;
    
    return total;
    
    
def sumOfOddPlace(cardNumber) {
    total = 0;
    for i in  range(len(cardNumber) - 1, 0, -2):
        total += cardNumber[i];
    }
    return total;
  }
  
def isValidCard(cardNumber):
    sumOfDoubleEvenPlace = sumOfDoubleEvenPlace(cardNumber);
    sumOfOddPlace = sumOfOddPlace(cardNumber);
    totalSum = sumOfDoubleEvenPlace + sumOfOddPlace;

    return totalSum % 10 == 0;
  
 

String cardNumber = input("Enter credit card number: ");

if (isValidLength(cardNumber)):
    cardType = determineCardType(cardNumber);
    isValid = isValidCard(cardNumber);


    print("*******************************************************");
    print(f"**Card Type: {cardType}", );
    print(f"**Card Number: {cardNumber}", );
    print(f"**Card Length: {len(cardNumber)}", );
    print(f"**Card Validity: {'Valid' if isValid else 'Invalid'}");
    print("*******************************************************");
else:
    print("Invalid card number length. Must be between 13 and 16 digits.");



  
